EventListener/SecurityListener.php

```diff
@@ -65,7 +66,12 @@ public function onKernelController(FilterControllerEvent $event)
         }
 
         if (!$this->language->evaluate($configuration->getExpression(), $this->getVariables($request))) {
-            throw new AccessDeniedException(sprintf('Expression "%s" denied access.', $configuration->getExpression()));
+            if ($statusCode = $configuration->getStatusCode()) {
+                throw new HttpException($statusCode, $configuration->getMessage());
+            }
+
+            $message = $configuration->getMessage() ?: sprintf('Expression "%s" denied access.', $configuration->getExpression());
```

Comment:
could be inlined.

Reply:
I would never use `Expression "%s" denied access.` as a message, but the more generic `Access denied` one.